lib/medusa.ex

```diff
@@ -147,39 +138,43 @@ defmodule Medusa do
     end
   end
 
-  defp validate_consume_function(function) when is_function(function) do
+  defp validate_consume_function(function) when is_function(function, 1) do
     validate_consume_function([function])
   end
 
   defp validate_consume_function([]) do
     :ok
   end
 
-  defp validate_consume_function([function|tail]) when is_function(function) do
-    case :erlang.fun_info(function, :arity) do
-      {:arity, 1} -> validate_consume_function(tail)
-      _ -> {:error, "arity must be 1"}
-    end
+  defp validate_consume_function([function|tail]) when is_function(function, 1) do
+    validate_consume_function(tail)
   end
 
   defp validate_consume_function(_) do
-    {:error, "consume must be function"}
+    Logger.error("consume must be function with arity 1")
+    {:error, "consume must be function with arity 1"}
   end
 
   defp do_validate_message([], _message) do
     :ok
   end
 
   defp do_validate_message([function|tail], %Message{} = message)
-  when is_function(function) do
+      when is_function(function, 1) do
     case apply(function, [message]) do
-      :ok -> do_validate_message(tail, message)
-      {:error, reason} -> {:error, reason}
-      reason -> {:error, reason}
+      :ok ->
+        do_validate_message(tail, message)
+      {:error, reason} ->
+        Medusa.Logger.error(message, "validation failed")
+        {:error, reason}
+      reason ->
```

Comment:
This whole clause makes me uncomfortable. how do we know the return value is an error for a reason if it does not come in an error tagged tuple. I think:
a) this clause should not be here
b) The error message should be "invalid return from validator, expected :ok or {:error, reason}"